### DISCUSSION TITLE
Validation with the import of other files

### DISCUSSION THREAD

**Discussion by mariolilliresearch** [eclipse-langium/langium#1217] [Q&A]
Hello,
I am working on a DSL for which it is necessary to use a file that contains the definition of all the functions, which will then be implemented in all subsequent language files. The problem I am trying to solve is to carry out validation of a file using the information present in the functions defined in that library file. Reading the documentation, the AST I'm working on contains only the nodes of the open file, and it should be necessary to use IndexManager to access the other files in the global scope. Is this correct? Could you give me an example of a validation function that can access another file that is not currently open? I would be extremely grateful.

**Comment by msujew** [comment]
@mariolilliresearch I am confused as to what your model looks like. Are you using cross references for the imported functions? In that case you can simply follow the cross reference.

If the references to your functions are implicit (which I would recommend against), you can ask for the document via the `LangiumDocuments` service.
